graft/coreth/plugin/evm/eth_gossiper.go

```diff
@@ -47,48 +53,46 @@ func NewGossipEthTxPool(mempool *txpool.TxPool, registerer prometheus.Registerer
 		return nil, fmt.Errorf("failed to initialize bloom filter: %w", err)
 	}
 
+	pendingTxs := make(chan core.NewTxsEvent, pendingTxsBuffer)
+	sub := mempool.SubscribeTransactions(pendingTxs, true)
+	if sub == nil {
+		return nil, errSubscribing
+	}
+
 	return &GossipEthTxPool{
 		mempool:    mempool,
-		pendingTxs: make(chan core.NewTxsEvent, pendingTxsBuffer),
+		pendingTxs: pendingTxs,
+		sub:        sub,
 		bloom:      bloom,
 	}, nil
 }
 
 type GossipEthTxPool struct {
 	mempool    *txpool.TxPool
 	pendingTxs chan core.NewTxsEvent
+	sub        event.Subscription
 
-	bloom *gossip.BloomFilter
 	lock  sync.RWMutex
-
-	// subscribed is set to true when the gossip subscription is active
-	// mostly used for testing
-	subscribed atomic.Bool
-}
-
-// IsSubscribed returns whether or not the gossip subscription is active.
-func (g *GossipEthTxPool) IsSubscribed() bool {
-	return g.subscribed.Load()
+	bloom *gossip.BloomFilter
 }
 
-func (g *GossipEthTxPool) Subscribe(ctx context.Context) {
-	sub := g.mempool.SubscribeTransactions(g.pendingTxs, false)
-	if sub == nil {
-		log.Warn("failed to subscribe to new txs event")
-		return
-	}
-	g.subscribed.CompareAndSwap(false, true)
-	defer func() {
-		sub.Unsubscribe()
-		g.subscribed.CompareAndSwap(true, false)
-	}()
+// UpdateBloomFilter continuously listens for new pending transactions from the
+// mempool and adds them to the bloom filter. If the bloom filter reaches its
+// capacity, it is reset and all pending transactions are re-added.
+func (g *GossipEthTxPool) UpdateBloomFilter(ctx context.Context) {
```

Comment:
nit: maybe can be renamed`BloomFilterUpdateLoop` or `DispatchBloomUpdater` etc etc to indicate the continous loop.